src/main/java/com/google/devtools/build/lib/analysis/test/InstrumentedFilesCollector.java

```diff
@@ -479,7 +479,7 @@ private static Iterable<TransitiveInfoCollection> getAllNonToolPrerequisites(
 
   private static List<? extends TransitiveInfoCollection> attributeDependencyPrerequisites(
       Attribute attribute, RuleContext ruleContext) {
-    if (attribute.getType().getLabelClass() == LabelClass.DEPENDENCY) {
+    if (attribute.getType().getLabelClass() == LabelClass.DEPENDENCY && !attribute.isToolDependency()) {
```

Comment:
Why not do this in getAllNonToolPrerequisites?

Reply:
You are right, that makes more sense. Changed.